drivers/regulator/pmic_shell.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/drivers/regulator/consumer.h>
+
+LOG_MODULE_REGISTER(regulator_shell, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define GET_REGULATOR(node) DEVICE_DT_GET(node),
+
+static const struct device *supported_pmics[] = {
+	DT_FOREACH_STATUS_OKAY(regulator_pmic, GET_REGULATOR)
+	};
+static const struct device *pmic_dev;
+
+#define MAX_REGULATOR_VOLTAGE 5000000 /* Max voltage of 5V */
+
+static int cmd_pmic_en(const struct shell *sh, size_t argc, char **argv)
+{
+	struct onoff_client cli;
+
+	if (pmic_dev == NULL) {
+		shell_print(sh, "No PMIC device has been selected");
```

Comment:
You need to use `shell_error()` for error printouts.

Reply:
Thanks, I've update this accordingly